### PR TITLE
Fix borrowck wrong move error when deref Box

### LINKED ISSUE #3040: "Cannot move" error in IDE but legal in compiler.
## Environment

* **Intellij-Rust plugin version: 0.2.0.2110-182**
* **Rust 1.30.0**
* **IntelliJ IDEA 2018.2.3 (CE) build IC-182.433.46**
* **MacOS Mojave**

## Problem description

IJ reports a "Cannot move" error dereferencing a Box however this seems to be legal upon compilation (by IJ and command line)

## Steps to reproduce
```rust
fn get_eggs(amt: i32) -> Box<(i32, String)> {
    let eggs = (amt, String::from("eggs"));
    Box::new(eggs)
}

fn main() {
    let dozen = get_eggs(12);
    let (num, item) = *dozen;    <<<======== IDE says "Cannot move" error.
    println!("We got {} {}", num, item);
}
```
And yet it compiles:

```
/Users/jgriffith/.cargo/bin/cargo run --color=always --package rust-sandbox --bin rust-sandbox
   Compiling rust-sandbox v0.1.0 (/Users/jgriffith/git/rust-sandbox)
    Finished dev [unoptimized + debuginfo] target(s) in 0.43s
     Running `target/debug/rust-sandbox`
We got 12 eggs

Process finished with exit code 0
```

x

### PR DESCRIPTION
**Pull request by ortem**
Fixes #3040.

### CODE REVIEW

File: src/test/kotlin/org/rust/ide/inspections/borrowck/RsBorrowCheckerMovesTest.kt [hunk 1]
```diff
@@ -245,8 +245,18 @@ class RsBorrowCheckerMovesTest : RsInspectionsTestBase(RsBorrowCheckerInspection
         struct S<'a> { a: &'a mut T }
         impl<'a> S<'a> {
             fn foo(&mut self) {
-                bar(self.a)
+                bar(self.љa)
```
[the remark this answers is not in the document]
Comment (ortem) [reply src/test/kotlin/org/rust/ide/inspections/borrowck/RsBorrowCheckerMovesTest.kt]: Thanks, fixed!

File: src/main/kotlin/org/rust/lang/core/types/ty/Ty.kt [hunk 2]
```diff
@@ -157,3 +159,6 @@ fun Ty.isMovesByDefault(lookup: ImplLookup): Boolean =
         is TyTuple -> types.any { it.isMovesByDefault(lookup) }
         else -> lookup.isCopy(this).not()
     }
+
+fun Ty.isBox(project: Project): Boolean =
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/lang/core/types/ty/Ty.kt]: `project` can be extracted from `item`

Comment (ortem) [reply src/main/kotlin/org/rust/lang/core/types/ty/Ty.kt]: Done

**Pull request merged**